CoralCommon/src/MonitoringEvent.cpp

```diff
@@ -22,31 +22,6 @@ namespace coral
       {
       }
 
-      Record::Record( const Record& right )
-      {
-        m_id          = right.m_id;
-        m_time        = right.m_time;
-        m_source      = right.m_source;
-        m_type        = right.m_type;
-        m_description = right.m_description;
-        m_data        = const_cast<std::auto_ptr<Payload>&>(right.m_data); // Needed to avoid double delete
-      }
-
-      Record& Record::operator=( const Record& right )
-      {
-        if( this != &right )
-        {
-          m_id          = right.m_id;
-          m_time        = right.m_time;
-          m_source      = right.m_source;
-          m_type        = right.m_type;
-          m_description = right.m_description;
-          m_data        = const_cast<std::auto_ptr<Payload>&>(right.m_data); // Needed to avoid double delete
-        }
-
-        return *this;
-      }
-
```

Comment:
@guitargeek why to remove this code?

Reply:
That code was apparently written before move symanics were in place. The copy constructor for `auto_ptr` here acts like the move constructor for the unique_ptr [1]. This means that the `right` record is now invalid bevause it's `m_data` was reset to `nullptr`. A copy constructor where the other object is mutated and then invalid is dangerous, we have move constructors for that now (and the move constructor is even generated by default for us, that's why code is removed but not added).



[1] https://en.cppreference.com/w/cpp/memory/auto_ptr/operator%3D